pkg/storage/segments/store.go

```diff
@@ -133,7 +133,17 @@ func (s *segmentStore) Put(ctx context.Context, data io.Reader, expiration time.
 			Metadata:       metadata,
 		}
 	} else {
-		limits, rootPieceID, err := s.metainfo.CreateSegment(ctx, "", "", -1, redundancy, s.maxEncryptedSegmentSize, expiration) // bucket, path and segment index are not known at this point
+		// early call to get bucket name, rest of the path cannot be determine yet
+		p, _, err := segmentInfo()
+		if err != nil {
+			return Meta{}, Error.Wrap(err)
+		}
+		bucket, _, _, err := split(p)
```

Comment:
`split` probably should have a nicer name: `splitPathFragments` or `splitPathTokens` or similar.

Reply:
refactored to `splitPathFragments`